camel/toolkits/function_tool.py

```diff
@@ -364,6 +479,72 @@ def set_parameter(self, param_name: str, value: Dict[str, Any]):
             param_name
         ] = value
 
+    def generate_openai_tool_schema(
+        self,
+        max_retries: int,
+        schema_assistant: Optional[BaseModelBackend] = None,
+    ) -> Dict[str, Any]:
+        r"""Generates an OpenAI tool schema for the specified function.
+
+        This method uses a language model (LLM) to generate the OpenAI tool
+        schema for the specified function by first generating a docstring and
+        then creating a schema based on the function's source code. If no LLM
+        is provided, it defaults to initializing a GPT_4O_MINI model. The
+        schema generation and validation process is retried up to
+        `max_retries` times in case of failure.
+
+
+        Args:
+            max_retries (int): The maximum number of retries for schema
+                generation and validation if the process fails.
+            schema_assistant (Optional[BaseModelBackend]): An optional LLM
+                backend model used for generating the docstring and schema. If
+                not provided, a GPT_4O_MINI model will be created.
+
+        Returns:
+            Dict[str, Any]: The generated OpenAI tool schema for the function.
+
+        Raises:
+            ValueError: If schema generation or validation fails after the
+            maximum number of retries, a ValueError is raised,
+            prompting manual schema setting.
+        """
+        if not schema_assistant:
+            logger.warning(
+                "Warning: No model provided. "
+                "Use GPT_4O_MINI to generate the schema."
```

Comment:
```suggestion
                f"Use `{ModelType.GPT_4O_MINI.value}` to generate the schema."
```

Reply:
Are you expecting to get the string “gpt-4o-mini” from `ModelType.GPT_4O_MINI.value`?
I tried the following:
print(f"Use `{ModelType.GPT_4O_MINI.value}` to generate the schema.")
But it returned:
> Use `ModelType.GPT_4O_MINI` to generate the schema.
I also tried several other approaches to extract the value from ModelType.GPT_4O_MINI, but none of them worked. Do you have any suggestions?

Reply:
This may be related to the recent refactoring. @WHALEEYE do you have any idea?
```
model_type = ModelType.GPT_4O_MINI

print(model_type.value)
print(str(model_type))
print(model_type.name)

unified_model_type = UnifiedModelType(model_type)
print(unified_model_type.value)
print(str(unified_model_type))
print(unified_model_type.name)
```
Output
```
ModelType.GPT_4O_MINI
ModelType.GPT_4O_MINI
GPT_4O_MINI
ModelType.GPT_4O_MINI
ModelType.GPT_4O_MINI
GPT_4O_MINI
```

How do we actually get the value out for sending to the client?

Reply:
@Zhangzeyu97 this issue has been fixed by @WHALEEYE https://github.com/camel-ai/camel/pull/1106. Can you merge the master and try again. Sorry for the bug.

Reply:
Thanks. It works now.